Null Hypothesis as Ho: Independence of categorical variables (customer order forms defective % does not varies by centre) 


Alternative hypothesis as Ha: Dependence of categorical variables (customer order forms defective % varies by centre)



In [1]:
import pandas as pd
from scipy import stats
from scipy.stats import norm, chi2_contingency
import numpy as np

In [2]:
df = pd.read_csv("D:\DS\Hypothesis testing\Costomer+OrderForm.csv")
df.head()

Phillippines   Indonesia       Malta       India
0   Error Free  Error Free   Defective  Error Free
1   Error Free  Error Free  Error Free   Defective
2   Error Free   Defective   Defective  Error Free
3   Error Free  Error Free  Error Free  Error Free
4   Error Free  Error Free   Defective  Error Free

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 300 entries, 0 to 299
Data columns (total 4 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   Phillippines  300 non-null    object
 1   Indonesia     300 non-null    object
 2   Malta         300 non-null    object
 3   India         300 non-null    object
dtypes: object(4)
memory usage: 9.5+ KB


In [4]:
df.describe()

Phillippines   Indonesia       Malta       India
count           300         300         300         300
unique            2           2           2           2
top      Error Free  Error Free  Error Free  Error Free
freq            271         267         269         280

In [5]:
df.Phillippines.value_counts()


Error Free    271
Defective      29
Name: Phillippines, dtype: int64

In [6]:
df.India.value_counts()


Error Free    280
Defective      20
Name: India, dtype: int64

In [7]:
df.Indonesia.value_counts()


Error Free    267
Defective      33
Name: Indonesia, dtype: int64

In [8]:
df.Malta.value_counts()


Error Free    269
Defective      31
Name: Malta, dtype: int64

In [9]:
# Make contingency array

cont_array = np.array([[271,267,269,280],[29,33,31,20]])
cont_array


array([[271, 267, 269, 280],
       [ 29,  33,  31,  20]])

In [10]:
p_value = chi2_contingency(cont_array)  # o/p is (Chi2 stats value, p_value, df, expected obsevations)
p_value

(3.858960685820355,
 0.2771020991233135,
 3,
 array([[271.75, 271.75, 271.75, 271.75],
        [ 28.25,  28.25,  28.25,  28.25]]))

In [11]:
p_value[1]

0.2771020991233135

Conclusion: As (p_value = 0.2771) > (α = 0.05); Accept Null Hypthesis i.e. Independence of categorical variables Thus, customer order forms defective % does not varies by centre.

